lib/fetch-server-test.js

```diff
@@ -0,0 +1,155 @@
+import isFunction from 'lodash/isFunction';
+import intersection from 'lodash/intersection';
+import difference from 'lodash/difference';
+
+import * as sinon from 'sinon';
+
+const VALID_METHODS = [`GET`, `POST`, `PATCH`, `PUT`, `DELETE`];
+
+/**
+ * Utility class to stub fetch requests with sinon. Instantiate with a set of
+ * route handlers mapping URL (sub)strings to canned response values, or functions
+ * which return canned responses/response values. Values will be serialized as JSON
+ * and returned in a Fetch-like Promise object.
+ *
+ * @example
+ * const fetchServer = new FetchServer({
+ *   '/foo.json': 'bar',
+ *   '/baz/quux': (url, params) => !!params.foo ? 'a' : 'z',
+ * }, {debug: true});
+ * fetchServer.start();
+ * const res = await fetch('https://myserver/foo.json');
+ * const val = await res.json(); // 'bar'
+ * 
+ * @example 
+ * You can also have the following handler types where key is any of the HTTP methods
+ * const handler = {
+ *   '/foo/bar`: {
+ *      GET: {
+ *        'content-type': 'application/json',
+ *        'status': 200,
+ *        'result': {}
+ *      },
```

Comment:
I think the ideal interface would allow
```
 * const handler = {
 *   '/foo/bar`: {
 *      GET: <fixture data>,
```
or
```
 * const handler = {
 *   '/foo/bar`: {
 *      GET: {
 *        'content-type': 'application/json',
 *        'status': 200,
 *        'result': <fixture data>
 *      },
```
or
```
 * const handler = {
 *   '/foo/bar`: {
 *      GET: [async] (url, params) => <fixture data>,
```
or
```
 * const handler = {
 *   '/foo/bar`: {
 *      GET: {
 *        'content-type': 'application/json',
 *        'status': 200,
 *        'result': [async] (url, params) => <fixture data>
 *      },
```
Let's discuss whether this is possible on the zooms. Also wondering if `body` would make more sense than `result` as key here